src/main/kotlin/com/ajaxproject/telegrambot/bot/service/TelegramService.kt

```diff
@@ -3,31 +3,39 @@ package com.ajaxproject.telegrambot.bot.service
 import com.ajaxproject.telegrambot.bot.sender.BankInfoBotSender
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
-import org.telegram.telegrambots.meta.api.methods.botapimethods.BotApiMethodMessage
+import org.telegram.telegrambots.meta.api.methods.BotApiMethod
 import org.telegram.telegrambots.meta.api.methods.send.SendMessage
+import org.telegram.telegrambots.meta.api.methods.updatingmessages.DeleteMessage
+import org.telegram.telegrambots.meta.api.objects.Message
 import org.telegram.telegrambots.meta.api.objects.replykeyboard.ReplyKeyboard
 
 @Component
 class TelegramService(
     val botSender: BankInfoBotSender,
 ) {
 
-    fun sendMessage(chatId: Long, text: String) {
-        sendMessage(chatId, text, null)
-    }
-
-    fun sendMessage(chatId: Long, text: String, replyKeyboard: ReplyKeyboard?) {
+    fun sendMessage(chatId: Long, text: String, replyKeyboard: ReplyKeyboard? = null): Message {
         val sendMessage = SendMessage.builder().also { msg ->
             msg.text(text)
             msg.chatId(chatId)
             msg.replyMarkup(replyKeyboard)
         }.build()
-        execute(sendMessage)
+        val responseMessage = execute(sendMessage)
+        return responseMessage!!
+    }
+
+    fun deleteMessage(chatId: Long, messageId: Int) {
+        val deleteMessage = DeleteMessage.builder().also { msg ->
+            msg.messageId(messageId)
+            msg.chatId(chatId)
+        }.build()
+        botSender.execute(deleteMessage)
     }
 
-    private fun execute(botApiMethod: BotApiMethodMessage) {
-        botApiMethod.runCatching { botSender.execute(botApiMethod) }
-            .onFailure { log.error("Failed to execute bot method={}", botApiMethod, it) }
+    private fun execute(botApiMethod: BotApiMethod<Message>): Message? {
+        return botApiMethod.runCatching { botSender.execute(botApiMethod) }
+            .onFailure { log.error("Failed to execute bot method {}", botApiMethod, it) }
+            .getOrNull()
```

Comment:
so, the function could return `null`. But above u invoke this function
```
        val responseMessage = execute(sendMessage)
        return responseMessage!!
```

So, you can get NPE. It's better to rethrow the original exception rather than getting NPE.

Please, use `!!` operator only in cases when you 100% sure a variable cannot be null.

As alternative you can process nullability in the next way:
```kotlin
val nullable: String = null
return nullable ?: throw MuCustomBusinessException("<what went wrong msg>")
```